README
staticgmap
==========

staticgmap is a lightweight tool to create static Google maps.

More information and examples: http://tools.s2studio.co.uk/staticgmap/

License: http://tools.s2studio.co.uk/staticgmap/#license

Copyright (c) 2012 Sarunas Savicianskas - sarunas@s2studio.co.uk


How to use
----------

To get started, download the plugin, unzip it and copy files to your website/application directory.
Load files in the <head> section of your HTML document. Make sure you also add the jQuery library.

    <head>
        <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7/jquery.min.js"></script>
        <script type="text/javascript" src="/staticgmap/jquery.staticgmap.1.0.1.js"></script>
    </head>

Create a html element, and add an id:

    <div id="element">This is where the static Google map should be generated by jQuery plugin!</div>

Initialise the script like this:

    <script>
        $(document).ready(function() {
            $('#element').staticgmap();
        });
    </script>

May also be passed an optional options object which will extend the default values. Example:

    <script>
        $(document).ready(function() {
            $('#element').staticgmap({
                address : '39 Steventon Road, London, UK',
                width   : 400,
                height  : 300
            });
        });
    </script>

Advanced
--------

You can customise the staticgmap as much as you want by passing specific options. Example:

    $("#element").staticgmap({
        width   : 300,
        height  : 250,
        zoom    : 14,
        maptype : 'hybrid', // roadmap, satellite, terrain, hybrid
        address : 'Tower Bridge, United Kingdom', // address or coordinates
        atlText : 'This is a static Google map generated by jQuery'
    });

### API

It`s possible to update staticgmap programmatically:

    $("#element").staticgmap('update', {
        width   : 600,
        height  : 400,
        zoom    : 17,
        maptype : 'satellite', // roadmap, satellite, terrain, hybrid
        address : 'Tower Bridge, United Kingdom', // address or coordinates
        atlText : 'This is an updated static Google map generated by jQuery'
    });

There are several methods that allow you to interact with and manipulate staticgmap, example:

    // Destroy staticgmap:
    $("#element").staticgmap('destroy');


Bug tracker
-----------

Have a bug? Please create an issue on GitHub at https://github.com/s2sarunas/staticgmap/issues

CHANGELOG
staticgmap - Changelog
=========

### Version 1.0.1 - August 06, 2012

First release.

source/jquery.staticgmap.pack.js
/*!
 * staticgmap - jQuery Plugin
 * version: 1.0.1 (06/08/2012)
 * @requires jQuery v1.6 or later
 *
 * Examples at http://tools.s2studio.co.uk/staticgmap
 * License: http://tools.s2studio.co.uk/staticgmap/#license
 *
 * Copyright 2012 Sarunas Savicianskas - sarunas@s2studio.co.uk
 *
 */(function($){var d={init:function(b){var c=$.extend({'width':300,'height':250,'zoom':14,'maptype':'hybrid','address':'London, UK','atlText':''},b);return this.each(function(){var a=$(this),data=a.data('staticgmap'),staticgmap=$('<img />',{'class':'staticgmap',src:'http://maps.google.com/maps/api/staticmap?zoom='+c.zoom+'&size='+c.width+'x'+c.height+'&maptype='+c.maptype+'&markers=size:normal|color:blue|'+c.address+'&sensor=false',css:{'height':c.height+'px','width':c.width+'px'},alt:c.altText});if(!data){$(this).data('staticgmap',staticgmap)}$(this).data('defaultvalue',$(this).html());$(this).html(staticgmap)})},destroy:function(){return this.each(function(){var a=$(this),data=a.data('staticgmap');$(window).unbind('.staticgmap');a.removeData('staticgmap');$(this).html($(this).data('defaultvalue'))})},update:function(b){var c=$.extend({},$(this).options,b);return this.each(function(){var a=$(this),staticgmap=$('<img />',{'class':'staticgmap',src:'http://maps.google.com/maps/api/staticmap?zoom='+c.zoom+'&size='+c.width+'x'+c.height+'&maptype='+c.maptype+'&markers=size:normal|color:blue|'+c.address+'&sensor=false',css:{'height':c.height+'px','width':c.width+'px'},alt:c.altText});$(this).data('staticgmap',staticgmap);$(this).html(staticgmap)})}};$.fn.staticgmap=function(a){if(d[a]){return d[a].apply(this,Array.prototype.slice.call(arguments,1))}else if(typeof a==='object'||!a){return d.init.apply(this,arguments)}else{$.error('Method '+a+' does not exist on jQuery.staticgmap')}}})(jQuery);

source/jquery.staticgmap.js
/*!
 * staticgmap - jQuery Plugin
 * version: 1.0.1 (06/08/2012)
 * @requires jQuery v1.6 or later
 *
 * Examples at http://tools.s2studio.co.uk/staticgmap
 * License: http://tools.s2studio.co.uk/staticgmap/#license
 *
 * Copyright 2012 Sarunas Savicianskas - sarunas@s2studio.co.uk
 *
 */


(function( $ ) {
    
    var methods = {
        init : function( options ) {
            // Create some defaults, extending them with any options that were provided
            var settings = $.extend( {
              'width'   : 300,
              'height'  : 250,
              'zoom'    : 14,
              'maptype' : 'hybrid', // roadmap, satellite, terrain, hybrid
              'address' : 'London, UK', // address or coordinates
              'atlText' : ''
            }, options);

            return this.each(function(){
                var $this = $(this),
                data = $this.data('staticgmap'),
                staticgmap = $('<img />', {
                    'class' : 'staticgmap',
                    src     : 'http://maps.google.com/maps/api/staticmap?zoom=' 
                        + settings.zoom + '&size=' + settings.width + 'x' 
                        + settings.height + '&maptype=' + settings.maptype 
                        + '&markers=size:normal|color:blue|' 
                        + settings.address + '&sensor=false',
                    css     : {
                            'height'   : settings.height + 'px',
                            'width'    : settings.width + 'px'
                    },
                    alt     : settings.altText
                });

                // If the plugin hasn't been initialized yet
                if ( ! data ) {
                    $(this).data('staticgmap', staticgmap);
                }
                
                $(this).data('defaultvalue', $(this).html());
                $(this).html(staticgmap);
            });
        },
        destroy : function( ) {
            return this.each(function(){
                var $this = $(this),
                data = $this.data('staticgmap');

                // Namespacing FTW
                $(window).unbind('.staticgmap');
                $this.removeData('staticgmap');
                
                $(this).html($(this).data('defaultvalue'));
            })
        },
        update : function( options ) { 
            // Create some defaults, extending them with any options that were provided
            var settings = $.extend( {}, $(this).options, options);

            return this.each(function(){
                var $this = $(this),
                staticgmap = $('<img />', {
                    'class' : 'staticgmap',
                    src     : 'http://maps.google.com/maps/api/staticmap?zoom=' 
                        + settings.zoom + '&size=' + settings.width + 'x' 
                        + settings.height + '&maptype=' + settings.maptype 
                        + '&markers=size:normal|color:blue|' 
                        + settings.address + '&sensor=false',
                    css     : {
                        'height'   : settings.height + 'px',
                        'width'    : settings.width + 'px'
                    },
                    alt     : settings.altText
                });
                
                // update current static maps with updated ones
                $(this).data('staticgmap', staticgmap);
                $(this).html(staticgmap);
            });
        }
    };    
    
    $.fn.staticgmap = function( method ) {
        if ( methods[method] ) {
            return methods[method].apply( this, Array.prototype.slice.call( arguments, 1 ));
        } else if ( typeof method === 'object' || ! method ) {
           return methods.init.apply( this, arguments );
        } else {
            $.error( 'Method ' +  method + ' does not exist on jQuery.staticgmap' );
        }
    };
})( jQuery );